sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -11,23 +11,25 @@
 using Azure.Core;
 using Azure.Messaging.ServiceBus.Core;
 using Azure.Messaging.ServiceBus.Diagnostics;
+using Azure.Messaging.ServiceBus.Primitives;
 using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Encoding;
 using Microsoft.Azure.Amqp.Framing;
 
 namespace Azure.Messaging.ServiceBus.Amqp
 {
     /// <summary>
     ///   A transport client abstraction responsible for brokering operations for AMQP-based connections.
-    ///   It is intended that the public <see cref="ServiceBusProcessorClient" /> make use of an instance
+    ///   It is intended that the public <see cref="ServiceBusReceiver" /> make use of an instance
     ///   via containment and delegate operations to it.
     /// </summary>
     ///
-    /// <seealso cref="Azure.Messaging.ServiceBus.Core.TransportConsumer" />
+    /// <seealso cref="Azure.Messaging.ServiceBus.Core.TransportReceiver" />
     ///
-    internal class AmqpConsumer : TransportConsumer
+    internal class AmqpReceiver : TransportReceiver
     {
         /// <summary>The default prefetch count to use for the consumer.</summary>
-        private const uint DefaultPrefetchCount = 0;
+        private const int DefaultPrefetchCount = 0;
```

Comment:
I'd  recommend leaving this as `uint` unless you've got a reason not to.   The track one implementation pushes this around as `int` only to end up converting it when it gets used with the link.   _(see [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageReceiver.cs#L243))_

Reply:
This was based on a comment from Krzysztof.

Reply:
I don't feel strongly, but am curious as to the rationale.   It's an internal value within this class, so any public-facing concerns would be moot.  Why not use the native type that gets sent and avoid the cast?

Reply:
The public type defined on the Options needs to be int. This can be uint, but we still have to do the cast at some point.

Reply:
No argument on that, but the cast here would potentially happen multiple times versus a single time when creating this class.   I don't think its a major concern, just seems to be an area for an easy efficiency.

Reply:
I guess it would happen multiple times in the case of retries, but there would generally only ever be one call to CreateReceivingLinkAsync per client, so it seemed more straightforward to just keep things consistent in terms of types.

Reply:
Remember that `CreateReceivingLink` is potentially called multiple times behind the scenes as well, such as when the link/session is abandoned due to inactivity or timeouts or there is a transient issue with network connectivity.   That's handled for you by the `FaultTolerantAmqpObject` which calls in to create the link on your behalf.

Reply:
Ok you've convinced me 😀